Sources/BSON/ExtendedJSONDecoder.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+/// Decoder for ExtendedJSON
+public class ExtendedJSONDecoder {
+    /// Initialize a Decoder
+    public init() {
+        fatalError("unimplemented")
+    }
+
+    /// Takes in JSON as `Data` encoded with `.utf8` and runs it through a `JSONDecoder` to get an
```

Comment:
I think this is an implementation detail we don't need to expose to users in our public documentation. the existence of a `JSON` type should be invisible to them entirely, and they don't really need to know how we got it. 

instead, we should state what this method's purpose is from a user's perspective: something like "Decodes an instance of the requested type `T` from the provided extended JSON data."

that said, the comment is definitely useful to a developer working on this library, so perhaps you could move it into the top of the method body?

Reply:
okay! done :)